test/fail_compilation/ice15441.d

```diff
@@ -1,8 +1,8 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/ice15441.d(24): Error: variable ice15441.main.__front$n$ type `void` is inferred from initializer `__r$n$.front()`, and variables cannot be of type `void`
-fail_compilation/ice15441.d(24): Error: expression `__r$n$.front()` is `void` and has no value
+fail_compilation/ice15441.d(24): Error: variable `ice15441.main.__front3` type `void` is inferred from initializer `__r2.front()`, and variables cannot be of type `void`
+fail_compilation/ice15441.d(24): Error: expression `__r2.front()` is `void` and has no value
```

Comment:
This change troubled me.  I don't understand what I did in dsybmol.d to cause this, and I'm not sure what's right.

Reply:
`$n$` is a d_do_test placeholder for any number to avoid having to adjust error messages whenever the compiler generates different unique numbers due to changes elsewhere. The test suite should pass with `$n$`, too.

Reply:
I wonder if the tick marks interfere with the d_do_test's text substitution.

Reply:
It's agnostic to anything but `$n$`: https://github.com/dlang/dmd/blob/master/test/d_do_test.d#L447

Reply:
Thanks @rainers, `$n$` is back in.